laurent22/joplin
# Mobile: Fixes #9123: Fix encryption when a resource doesn't have an associated file

# Summary

When `Resource.ts` expects file not found errors to have code `ENOENT`. Instead, on mobile `fsdriver-rn`'s `readFileChunk` threw a `TypeError: Cannot read property "size" of null`.

This pull request corrects `readFileChunk` to throw an exception with the correct code.

Fixes #9123.

# Relevant files

https://github.com/laurent22/joplin/blob/7cec62fc7144d06dd938f675ed9957eec47c2e83/packages/lib/models/Resource.ts#L227

https://github.com/laurent22/joplin/blob/7cec62fc7144d06dd938f675ed9957eec47c2e83/packages/lib/Synchronizer.ts#L40

# Testing

This pull request can be tested manually by
1. Applying [this diff](https://github.com/laurent22/joplin/issues/9123#issuecomment-1795953296) that adds a `TEST: Add resource without file` button to the mobile config screen
2. Setting up sync with encryption
3. Clicking the new button
4. Syncing

A `Some items cannot be synchronized` banner should be visible.

After this,
1. Disable encryption
2. Click the `TEST: Add resource without file` button
3. Resync

The new item should be added to the sync errors list with the message `TEST (uuidhere) could not be uploaded: File not found: /some/path/here/uuidhere.txt` (as before this PR).
